File: frontend/lib/models/crop_cycle.dart
class CropCycle {
  final int id;
  final int farmId;
  final String cropType;
  final DateTime startDate;
  DateTime? endDate;
  bool isActive;

  CropCycle({
    required this.id,
    required this.farmId,
    required this.cropType,
    required this.startDate,
    this.endDate,
    required this.isActive,
  });
}

File: src/main/java/com/farmer/entity/CropCycle.java
package com.farmer.entity;

import jakarta.persistence.*;
import java.time.LocalDate;
import com.farmer.entity.*;

import lombok.*;
@Setter@Getter@Builder
@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name="cropcycle")
public class CropCycle {
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name="farm_id",nullable=false)
    private Farm farm;

    @Column(nullable=false)
    private String cropName;

    @Column(nullable=false)
    private LocalDate startDate;

    @Column(nullable=true)
    private LocalDate endDate;

    @Column(nullable=false)
    private String  status;

    
}
